File: Backend/src/main/java/com/frozenleafstudio/dev/AutomatedSetlist/Playlist/AppTrack.java
package com.frozenleafstudio.dev.AutomatedSetlist.Playlist;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AppTrack {
    private boolean trackFound;
    private String songUri;
    private String songName;
    private String artistName;
    private String albumName;
    private String albumImageUrl;
    private String details;
    private boolean isTape;
    private boolean isCover;
}

File: Frontend/src/models/AppTrack.ts
export class AppTrack {
  trackFound: boolean = false;
  songUri: string = "";
  songName: string = "";
  artistName: string = "";
  albumName: string = "";
  albumImageUrl: string = "";
  details: string = "";
  tape: boolean = false;
  cover: boolean = false;

  constructor(initializer: Partial<AppTrack> = {}) {
    Object.assign(this, initializer);
  }
}
